homeassistant/components/israel_rail/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Config flow for israel rail."""
+
+import logging
+from typing import Any
+
+from israelrailapi import TrainSchedule
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_DESTINATION, CONF_START, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_START): cv.string,
+        vol.Required(CONF_DESTINATION): cv.string,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IsraelRailConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Israel rail config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Async user step to set up the connection."""
+        errors = {}
+        if user_input:
+            train_schedule = TrainSchedule()
+            try:
+                await self.hass.async_add_executor_job(
+                    train_schedule.query,
+                    user_input[CONF_START],
+                    user_input[CONF_DESTINATION],
+                )
+            except Exception:
+                _LOGGER.exception("Unknown error")
+                errors["base"] = "unknown"
+            if not errors:
+                unique_id = f"{user_input[CONF_START]} {user_input[CONF_DESTINATION]}"
```

Comment:
Can you give an example of the start and destination?

Reply:
These are station names of the railway. It's unique
like:
בית יהושע
תל אביב - סבידור מרכז

Reply:
Can we get a list of all stations someone can choose from?

Reply:
Yes I think that can be an option

Reply:
Done